src/controllers/forecast.mjs

```diff
@@ -26,14 +41,13 @@ export const getForecastForCity = async (req, res) => {
 
 export const getForecastForAllCities = async (_, res) => {
   try {
-    const citiesData = await getFileData("./data/cities.json");
+    const cities = await getFileData(CITIES_FILE_PATH);
+    cities.forEach(async (city) => await checkUpdate(city));
 
-    const forecasts = await Promise.all(
-      citiesData.map((city) => getForecastByCity(city))
-    );
-    res.status(200).json(forecasts);
-  } catch (error) {
-    console.error("Error getting forecasts for all cities:", error);
+    res.status(200).json(await getFileData(WEATHER_LOG_FILE_PATH));
```

Comment:
Due to main comment, we wouldn't need to read file one more time
Time saved, resources saved